lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -1835,202 +1836,204 @@ void HdVP2Material::Sync(
         VtValue vtMatResource = sceneDelegate->GetMaterialResource(id);
 
         if (vtMatResource.IsHolding<HdMaterialNetworkMap>()) {
-            auto* const param = static_cast<HdVP2RenderParam*>(_renderDelegate->GetRenderParam());
-            const bool  inUntexturedMode
-                = (param->GetDrawScene().GetDisplayStyle() & MHWRender::MFrameContext::kTextured)
-                == 0;
+            const HdMaterialNetworkMap& fullNetworkMap
+                = vtMatResource.UncheckedGet<HdMaterialNetworkMap>();
 
-            HdMaterialNetworkMap        untexturedNetworkMap;
-            const HdMaterialNetworkMap& networkMap = inUntexturedMode
-                ? untexturedNetworkMap
-                : vtMatResource.UncheckedGet<HdMaterialNetworkMap>();
+            // untextured network is always synced
+            HdMaterialNetworkMap untexturedNetworkMap = fullNetworkMap;
+            ConvertNetworkMapToUntextured(untexturedNetworkMap);
+            _compiledNetworks[kUntextured].Sync(sceneDelegate, untexturedNetworkMap);
 
-            if (inUntexturedMode) {
-                untexturedNetworkMap = vtMatResource.UncheckedGet<HdMaterialNetworkMap>();
-                ConvertNetworkMapToUntextured(untexturedNetworkMap);
+            // full network is synced only if required by display style
```

Comment:
Unclear why the untextured network is always synced. Especially when we want the full network for the textured Repr.

Reply:
The default viewport style in Maya is untextured, so I thought of untextured material network as default as well, especially since it's lightweight compared to the other one. However, you are right and technically we shouldn't be forced to always sync it.